api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3324,14 +3333,56 @@ public void makeBucket(String bucketName, String region)
    * @throws RegionConflictException thrown to indicate passed region conflict with default region.
    * @throws XmlParserException thrown to indicate XML parsing error.
    */
+  @Deprecated
   public void makeBucket(String bucketName, String region, boolean objectLock)
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, RegionConflictException,
           XmlParserException {
-    // If region param is not provided, set it with the one provided by constructor
-    if (region == null) {
+
+    MakeBucketArgs.Builder makeBucketArgs = new MakeBucketArgs.Builder();
+    makeBucketArgs.bucket(bucketName).region(region).objectLock(objectLock);
+    MakeBucketArgs args = makeBucketArgs.build();
+    this.makeBucket(args);
+  }
+
+  /**
+   * Creates a bucket makeBucketArgs.
+   *
+   * <pre>Example:{@code
+   * minioClient.makeBucket(MakeBucketArgs makebucketArgs);
+   * }</pre>
+   *
+   * @param args Object with bucket name, region and lock functionality
+   * @throws ErrorResponseException thrown to indicate S3 service returned an error response.
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws RegionConflictException thrown to indicate passed region conflict with default region.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public void makeBucket(MakeBucketArgs args)
+      throws InvalidBucketNameException, RegionConflictException, InsufficientDataException,
+          InternalException, InvalidResponseException, InvalidKeyException,
+          NoSuchAlgorithmException, XmlParserException, ErrorResponseException, IOException {
+
+    if ((args == null)) {
+      throw new IllegalArgumentException("null value is not allowed in arguments");
+    }
+
+    String region = null;
+    if (args.region() != null && !args.region().isEmpty()) {
+      region = args.region();
+    } else if (this.region != null && !this.region.isEmpty()) {
       region = this.region;
+    } else {
+      region = US_EAST_1;
     }
 
     // If constructor already sets a region, check if it is equal to region param if provided
```

Comment:
I am thinking to get rid of constructor driven `region` setting and have `BucketName` object coming with `region` value itself.  That could be a separate PR after this PR merged.